### PR TITLE
Adding all Auth Samples

### PR DESCRIPTION
**Pull request by peakyy**
Hi Will,

Here is the pull request. What I have done is made what each sample is referring to more obvious by changing folder names, etc. and adding the ADAL sample. The link to specific auth guidance is going to point to a page I'm working on that I have not yet pushed to prod and that's why it currently points to "microsoft.com". I just realized I can update the url to match what will be expected before we accept this pull request

              Peaky

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,9 +4,10 @@
 
 Samples that show how to auth with Visual Studio Team Services and Team Foundation Server.
 
-Learn more about [integrating with Team Services](https://www.visualstudio.com/docs/integrate/extensions/overview)
+Learn more about [integrating with Team Services](https://www.visualstudio.com/docs/integrate/extensions/overview) and [specific authentication guidence](https://www.microsoft.com)
```
Comment (justmarks) [inline README.md]: The auth guidance link needs to be updated

File: README.md [hunk 2]
```diff
@@ -4,9 +4,10 @@
...
 
 ## Samples
 
-* [ASP.NET web app](dotnet/aspnet/README.md)
+* [ASP.NET web app (OAuth Sample)](ASP.NET\ web\ app\ \(OAuth\ Sample\)/aspnet/README.md)
+* [Using REST APIs with ADAL (ADAL C# Sample)](Using\ REST\ APIs\ with\ ADAL\ \(ADAL\ C#\ Sample\)/README.md)
```
Comment (justmarks) [inline README.md]: these links seem odd, why are the spaces escaped?  Also, why is there a /?

File: Using REST APIs with ADAL (ADAL C# Sample)/Program.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                result = ctx.AcquireTokenAsync(VSTSResourceId, clientId, new Uri("urn:ietf:wg:oauth:2.0:oob"), new PlatformParameters(PromptBehavior.Auto)).Result;
+                Console.WriteLine("Token expires on: " + result.ExpiresOn);
+            }
+            catch (Exception ex)
+            {
+                throw ex.InnerException;
+            }
+
+            do
+            {
```
Comment (justmarks) [inline Using REST APIs with ADAL (ADAL C# Sample)/Program.cs]: this is a bad example, why are you looping over and over?  That was simply something I was doing as a test.  You should simplify to just call ListProjects once

File: Using REST APIs with ADAL (ADAL C# Sample)/Program.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+                var bearerAuthHeader = new AuthenticationHeaderValue("Bearer", result.AccessToken);
+
+                try
+                {
+                    ListProjects(VSTSAccountName, bearerAuthHeader);
+                    Thread.Sleep(300000); //Sleep for 5 minutes
+                }
+                catch (UnauthorizedAccessException uae)
+                {
```
Comment (justmarks) [inline Using REST APIs with ADAL (ADAL C# Sample)/Program.cs]: You should comment here on what happens when the token expires, are they given a UAE and does this work or should they do something else?

File: Using REST APIs with ADAL (ADAL C# Sample)/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Using REST APIs with Adal (Adal C# Sample)
+
+The Azure Active Directory Authentication Library (ADAL) enables client application developers to authenticate cloud users and obtain access tokens for API usage.
+
+## Sample Application
+
+This buildable sample will walk you through the steps to create a native console application which uses ADAL to authenticate a user and return a list of all projects inside a selected VSTS account.
```
Comment (justmarks) [inline Using REST APIs with ADAL (ADAL C# Sample)/README.md]: Native has multiple meanings so we need to be careful here.  I think should remove the term as native console sounds like it's a C++ app which this is not, it's managed :)  The intention of "native" was that the application is client side so let's use that instead in all locations

File: Using REST APIs with ADAL (ADAL C# Sample)/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# Using REST APIs with Adal (Adal C# Sample)
+
+The Azure Active Directory Authentication Library (ADAL) enables client application developers to authenticate cloud users and obtain access tokens for API usage.
```
Comment (justmarks) [inline Using REST APIs with ADAL (ADAL C# Sample)/README.md]: Let's add a link to the ADAL documentation

File: Using REST APIs with ADAL (ADAL C# Sample)/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+## Step 2: Install and configure ADAL (optional)
+
+Package: `Microsoft.Identity.Model.Clients.ActiveDirectory` has already been installed and configured in the sample, but if you are adding to your own project you will need to [install and configure it yourself](https://www.nuget.org/packages/Microsoft.IdentityModel.Clients.ActiveDirectory). 
+
+## Step 3: Run the sample
+
+1. Navigate to the cloned repo `Using REST APIs with ADAL (ADAL C# Sample)`
+2. Use [Nugget package restore](https://docs.microsoft.com/en-us/nuget/consume-packages/package-restore) to ensure you have the latest package installed
```
Comment (justmarks) [inline Using REST APIs with ADAL (ADAL C# Sample)/README.md]: one g in NuGet

File: Using REST APIs with ADAL (ADAL C# Sample)/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+git clone https://github.com/Microsoft/vsts-auth-samples.git
+```
+
+## Step 2: Install and configure ADAL (optional)
+
+Package: `Microsoft.Identity.Model.Clients.ActiveDirectory` has already been installed and configured in the sample, but if you are adding to your own project you will need to [install and configure it yourself](https://www.nuget.org/packages/Microsoft.IdentityModel.Clients.ActiveDirectory). 
+
+## Step 3: Run the sample
+
+1. Navigate to the cloned repo `Using REST APIs with ADAL (ADAL C# Sample)`
```
Comment (justmarks) [inline Using REST APIs with ADAL (ADAL C# Sample)/README.md]: That's not the repro name, that's the folder name

**Comment by justmarks** [comment]
@peakyy I don't like the folder name, it's too long.  Can we simplify to "Managed Client sample"

File: README.md (hunk 1 above)
Comment (peakyy) [inline README.md]: Updated the link to mirror what it will be when guidance page goes live.

File: README.md (hunk 2 above)
Comment (peakyy) [inline README.md]: changed folder structure to make it less awkward

File: Using REST APIs with ADAL (ADAL C# Sample)/Program.cs (hunk 3 above)
Comment (peakyy) [inline Using REST APIs with ADAL (ADAL C# Sample)/Program.cs]: removed the useless loop that existed in the code

File: Using REST APIs with ADAL (ADAL C# Sample)/Program.cs (hunk 4 above)
Comment (peakyy) [inline Using REST APIs with ADAL (ADAL C# Sample)/Program.cs]: What does actually happen in that case?

File: Using REST APIs with ADAL (ADAL C# Sample)/README.md (hunk 6 above)
Comment (peakyy) [inline Using REST APIs with ADAL (ADAL C# Sample)/README.md]: link added

File: Using REST APIs with ADAL (ADAL C# Sample)/README.md (hunk 5 above)
Comment (peakyy) [inline Using REST APIs with ADAL (ADAL C# Sample)/README.md]: updated here and in the auth_guidance page

File: Using REST APIs with ADAL (ADAL C# Sample)/README.md (hunk 8 above)
Comment (peakyy) [inline Using REST APIs with ADAL (ADAL C# Sample)/README.md]: updated

File: Using REST APIs with ADAL (ADAL C# Sample)/README.md (hunk 7 above)
Comment (peakyy) [inline Using REST APIs with ADAL (ADAL C# Sample)/README.md]: thank you, I didn't catch that one!

File: Client_OM_Sample/ClientOmConsoleApp/ClientOmConsoleApp/Program.cs [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        //Console application to execute a user defined work item query
+        static void Main(string[] args)
+        {
+            //=========== Config [edit these] ===================
+            string collectionUri = "https://peakyyy.visualstudio.com";
+            Wiql query = new Wiql() { Query = "Select [State], [Title] from WorkItems where [Work Item Type] = 'Bug' And [Tags] Contains 'findMe'" };
+            //===================================================
+
+            //Prmot user for credential for collection specified above
```
Comment (justmarks) [inline Client_OM_Sample/ClientOmConsoleApp/ClientOmConsoleApp/Program.cs]: typo "Prompt"

File: Client_OM_Sample/ClientOmConsoleApp/ClientOmConsoleApp/Program.cs [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+            VssCredentials cred = new VssClientCredentials(false);
+            cred.PromptType = CredentialPromptType.PromptIfNeeded;
```
Comment (justmarks) [inline Client_OM_Sample/ClientOmConsoleApp/ClientOmConsoleApp/Program.cs]: can we add comments on what this means?  Is there any doc to link to about these options?

File: Device_Profile_Sample/Device_Profile_Get_Identity_Data/Program.cs [hunk 11]
```diff
@@ -0,0 +1,141 @@
+﻿using Microsoft.IdentityModel.Clients.ActiveDirectory;
```
Comment (justmarks) [inline Device_Profile_Sample/Device_Profile_Get_Identity_Data/Program.cs]: This is just a copy of the sample from ADAL - I want it to be specific to VSTS, aka call a REST API or client library call like the other samples.  Remove the cruft like the colors, comments, command line, etc. and make this match the other samples

File: Device_Profile_Sample/Device_Profile_Get_Identity_Data/README.md [hunk 12]
```diff
@@ -0,0 +1,13 @@
+# Device Profile Sample
+
+For a headless text output client application, it is not possible authenticate through an interactive prompt. Instead a text only approach is necessary. This flow leverages a user's external device (i.e. phone) to authenticate through an interactive login prompt and pass the auth token to the headless application. For more information [click here](https://azure.microsoft.com/en-us/resources/samples/active-directory-dotnet-deviceprofile/?v=17.23h).
+
+## Sample Application
+
+This buildable sample will walk you through the steps to create a client-side console application which uses ADAL to authenticate a user via the [Device Profile flow](https://azure.microsoft.com/en-us/resources/samples/active-directory-dotnet-deviceprofile/?v=17.23h) and returns a JSON string containing all VSTS identity data for the authenticated user.
+
+## Step 1: Clone or download vsts-auth-samples repository
```
Comment (justmarks) [inline Device_Profile_Sample/Device_Profile_Get_Identity_Data/README.md]: where are the rest of the steps?

File: Using REST APIs with ADAL (ADAL C# Sample)/Program.cs (hunk 3 above)
Comment (justmarks) [inline Using REST APIs with ADAL (ADAL C# Sample)/Program.cs]: why is there a loop at all?

File: Device_Profile_Sample/Device_Profile_Get_Identity_Data/Program.cs (hunk 11 above)
Comment (peakyy) [inline Device_Profile_Sample/Device_Profile_Get_Identity_Data/Program.cs]: I removed all of the bloat code and call an api endpoint which prints team project info, so it should match the other samples.

File: Device_Profile_Sample/Device_Profile_Get_Identity_Data/README.md (hunk 12 above)
Comment (peakyy) [inline Device_Profile_Sample/Device_Profile_Get_Identity_Data/README.md]: Finished this

File: Client_OM_Sample/ClientOmConsoleApp/ClientOmConsoleApp/Program.cs (hunk 9 above)
Comment (peakyy) [inline Client_OM_Sample/ClientOmConsoleApp/ClientOmConsoleApp/Program.cs]: fixed

File: Client_OM_Sample/ClientOmConsoleApp/ClientOmConsoleApp/Program.cs (hunk 10 above)
Comment (peakyy) [inline Client_OM_Sample/ClientOmConsoleApp/ClientOmConsoleApp/Program.cs]: I couldn't find any, I just tried each one until the sample worked to be honest.

Comment (peakyy) [inline Client_OM_Sample/ClientOmConsoleApp/ClientOmConsoleApp/Program.cs]: I did find them to be relatively self explanatory

**Pull request merged**